Yafc/Workspace/ProductionTable/ProductionTableView.cs

```diff
@@ -692,6 +692,7 @@ private enum ProductDropdownType {
                 recipe.RecordUndo().fuel = fuel;
             });
             var allProduction = goods == null ? Array.Empty<Recipe>() : variants == null ? goods.production : variants.SelectMany(x => x.production).Distinct().ToArray();
+            var fuelUseList = goods == null ? Array.Empty<Recipe>() : goods.fuelFor.AsEnumerable().OfType<EntityCrafter>().SelectMany(e => e.recipes).OfType<Recipe>().Distinct().OrderBy(e => e, DataUtils.DefaultRecipeOrdering).ToArray();
```

Comment:
I'd suggest expanding this ternary into an `if` too and also refactoring adjacent ternaries into `if`s. You can check it yourself how hard it is to read them in the section Files Changed of the PR.

Reply:
I didnt touch `allProduction`, but after removing the ternary i'm scratching my head if this was intended:

```C#
Recipe[] allProduction;
if (goods == null)
{
  allProduction = [];
}
else
{
  if (variants == null)
  {
    allProduction = goods.production;
  }
  else
  {
    allProduction = variants.SelectMany(x => x.production).Distinct().ToArray();
  }
}
```

I've now refactored it to this:


```C#
Recipe[] allProduction = variants.SelectMany(x => x.production).Distinct().ToArray() ?? goods.production ?? [];
```
Much more readable, but not the same code. I wonder if this fixes or introduces a bug.

Reply:
> I wonder if this fixes or introduces a bug.

You can make unit tests and validate both versions. The commit order would be: first unit tests, then the change.

Reply:
That would suggest i know the intended behavior of the code.

I can tell from the change i made that it works different now: Before if `goods` was null, it just returned an empty list.
If `goods` was not null, it would prefer to use the result of `variants.SeectMany(...)`, but if `variants` was `null`, it fell back to `goods`.

This feels like a bug, and there should have been different brackets in the original ternary so it would do: Try variants, if null, try goods, if null, use empty array. That's how i changed it, and some quick tests show i didn't break anything. But a real test would be to understand when `variants` is filled, and if it's possible to fill `variants` while `goods` is `null`. If that scenario is possible, then decide if it would actually be preferred to use variants then, or use an empty list.

I'll dig a bit more trough the code and see if i can find that last scenario.


Re unit testing: yeah i should really start doing some of that....

Reply:
Am i correct that this change is holding back this Pr? Or should i rebase it?

If this change is holding it back, i rather revert to the old `allProduction` line and test or refactor it in a separate issue.

Reply:
> Am i correct that this change is holding back this Pr?

You are correct. I was waiting because the following line was in your previous message:

> I'll dig a bit more trough the code and see if i can find that last scenario.

Since we both need to understand what this code did and if we changed it correctly, then putting it into a separate PR is indeed a good idea.